# HG changeset patch
# User Rob Mahurin <rob@utk.edu>
# Date 1235548047 -3600
#      Wed Feb 25 08:47:27 2009 +0100
# Branch release-3-0-x
# Node ID 0d7e145c1ca4b07c7c9d974ae59762acc37a10bd
# Parent  a24565131108d8ec670d656799c4d103690e938d
fix order of legends

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-02-24  Rob Mahurin  <rob@utk.edu>	
+
+	* plot/legend.m: Fix legend order.  Add explicit colors to demos.
+	From Marco Caliari <marco.caliari@univr.it>
+	
 2009-02-23  John W. Eaton  <jwe@octave.org>
 
 	* plot/fplot.m: Fix nargin check.
 	From Joel Parker <Joel.Parker@radiancetech.com>.
 
 2008-01-15  Rafael Laboissiere  <rafael@debian.org>
 
 	* gethelp.cc: Include <cstdio>.
diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -166,17 +166,17 @@ function legend (varargin)
       endif
     endfor
     if (! have_data)
       warning ("legend: plot data is empty; setting key labels has no effect");
     endif
   endif
 
   warned = false;
-  for i = nargs:-1:1
+  for i = 1:nargs
     arg = varargin{i};
     if (ischar (arg))
       while (k <= nkids
 	     && ! (strcmp (get (kids(k), "type"), "line")
 		   || strcmp (get (kids(k), "type"), "surface")
 		   || strcmp (get (kids(k), "type"), "patch")))
 	k++;
       endwhile
@@ -196,29 +196,30 @@ function legend (varargin)
   if (turn_on_legend)
     set (ca, "key", "on");
   endif
 
 endfunction
 
 %!demo
 %! close all;
-%! plot(1:10, 1:10, 1:10, fliplr(1:10));
+%! plot(1:10, 1:10, 'b', 1:10, fliplr(1:10), 'g');
 %! title("incline is blue and decline is green");
 %! legend({"I'm blue", "I'm green"}, "location", "east")
 
 %!demo
 %! close all;
 %! plot(1:10, 1:10);
 %! title("a very long label can sometimes cause problems");
 %! legend({"hello world"}, "location", "northeastoutside")
 
 %!demo
 %! close all;
 %! labels = {};
+%! color = 'bgryk';
 %! for i = 1:5
-%!     plot(1:100, i + rand(100,1)); hold on;
+%!     plot(1:100, i + rand(100,1), color(i)); hold on;
 %!     labels = {labels{:}, strcat("Signal ", num2str(i))};
 %! endfor; hold off;
 %! title("Signals with random offset and uniform noise")
 %! xlabel("Sample Nr [k]"); ylabel("Amplitude [V]");
 %! legend(labels, "location", "southoutside")
 %! legend("boxon")
